pytest_logger/plugin.py

```diff
@@ -97,9 +97,20 @@ def pytest_runtest_teardown(self, item, nextitem):
         if logger:
             logger.on_teardown()
 
+    @pytest.mark.hookwrapper
     def pytest_runtest_makereport(self, item, call):
+        outcome = yield
+        tr = outcome.get_result()
         logger = getattr(item, '_logger', None)
         if logger:
+            if tr.outcome == 'failed' and self._logsdir:
+                failedlogsdir = self._logsdir.join('failedlogs')
+                failedlogsdir.ensure(dir=1)
+                nodeid = _sanitize_nodeid(item.nodeid)
+                nodepath = os.path.dirname(nodeid)
+                failedlogsdir.join(nodepath).ensure(dir=1)
+                destdir_relpath = os.path.relpath(self._logsdir.join(nodeid), failedlogsdir.join(nodepath))
```

Comment:
There is an error in this line, I didn't debug it very thoroughly, but you can reproduce this way:

Rebase on this commit (init script mentioned old python and pytest versions):
https://github.com/aurzenligl/pytest-logger/commit/016ab436e7fabb93603f2dd6ea252cb738e29e57
Then checkout `study`:
https://github.com/aurzenligl/study
And do this:
```
aurzenligl@dell7510 /tmp $ git clone https://github.com/aurzenligl/study
aurzenligl@dell7510 /tmp $ cd study/python-logger/
aurzenligl@dell7510 /tmp/study/python-logger $ source ~/projects/pytest-logger/init-tox-env 
(py35-pytest37) aurzenligl@dell7510 /tmp/study/python-logger $ pytest --error
(...)
test_p.py ...                                                                                                                                                                                                                          [ 30%]
test_x.py .
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/home/aurzenligl/projects/pytest-logger/.tox/py35-pytest37/lib/python3.5/site-packages/_pytest/main.py", line 178, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/home/aurzenligl/projects/pytest-logger/.tox/py35-pytest37/lib/python3.5/site-packages/_pytest/main.py", line 215, in _main
INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
INTERNALERROR>   File "/home/aurzenligl/projects/pytest-logger/.tox/py35-pytest37/lib/python3.5/site-packages/pluggy/hooks.py", line 289, in __call__
INTERNALERROR>     return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/home/aurzenligl/projects/pytest-logger/.tox/py35-pytest37/lib/python3.5/site-packages/pluggy/manager.py", line 68, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/home/aurzenligl/projects/pytest-logger/.tox/py35-pytest37/lib/python3.5/site-packages/pluggy/manager.py", line 62, in <lambda>
INTERNALERROR>     firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
INTERNALERROR>   File "/home/aurzenligl/projects/pytest-logger/.tox/py35-pytest37/lib/python3.5/site-packages/pluggy/callers.py", line 208, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/home/aurzenligl/projects/pytest-logger/.tox/py35-pytest37/lib/python3.5/site-packages/pluggy/callers.py", line 80, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/home/aurzenligl/projects/pytest-logger/.tox/py35-pytest37/lib/python3.5/site-packages/pluggy/callers.py", line 187, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/home/aurzenligl/projects/pytest-logger/.tox/py35-pytest37/lib/python3.5/site-packages/_pytest/main.py", line 236, in pytest_runtestloop
INTERNALERROR>     item.config.hook.pytest_runtest_protocol(item=item, nextitem=nextitem)
INTERNALERROR>   File "/home/aurzenligl/projects/pytest-logger/.tox/py35-pytest37/lib/python3.5/site-packages/pluggy/hooks.py", line 289, in __call__
INTERNALERROR>     return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/home/aurzenligl/projects/pytest-logger/.tox/py35-pytest37/lib/python3.5/site-packages/pluggy/manager.py", line 68, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/home/aurzenligl/projects/pytest-logger/.tox/py35-pytest37/lib/python3.5/site-packages/pluggy/manager.py", line 62, in <lambda>
INTERNALERROR>     firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
INTERNALERROR>   File "/home/aurzenligl/projects/pytest-logger/.tox/py35-pytest37/lib/python3.5/site-packages/pluggy/callers.py", line 208, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/home/aurzenligl/projects/pytest-logger/.tox/py35-pytest37/lib/python3.5/site-packages/pluggy/callers.py", line 80, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/home/aurzenligl/projects/pytest-logger/.tox/py35-pytest37/lib/python3.5/site-packages/pluggy/callers.py", line 187, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/home/aurzenligl/projects/pytest-logger/.tox/py35-pytest37/lib/python3.5/site-packages/_pytest/runner.py", line 66, in pytest_runtest_protocol
INTERNALERROR>     runtestprotocol(item, nextitem=nextitem)
INTERNALERROR>   File "/home/aurzenligl/projects/pytest-logger/.tox/py35-pytest37/lib/python3.5/site-packages/_pytest/runner.py", line 81, in runtestprotocol
INTERNALERROR>     reports.append(call_and_report(item, "call", log))
INTERNALERROR>   File "/home/aurzenligl/projects/pytest-logger/.tox/py35-pytest37/lib/python3.5/site-packages/_pytest/runner.py", line 163, in call_and_report
INTERNALERROR>     report = hook.pytest_runtest_makereport(item=item, call=call)
INTERNALERROR>   File "/home/aurzenligl/projects/pytest-logger/.tox/py35-pytest37/lib/python3.5/site-packages/pluggy/hooks.py", line 289, in __call__
INTERNALERROR>     return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/home/aurzenligl/projects/pytest-logger/.tox/py35-pytest37/lib/python3.5/site-packages/pluggy/manager.py", line 68, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/home/aurzenligl/projects/pytest-logger/.tox/py35-pytest37/lib/python3.5/site-packages/pluggy/manager.py", line 62, in <lambda>
INTERNALERROR>     firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
INTERNALERROR>   File "/home/aurzenligl/projects/pytest-logger/.tox/py35-pytest37/lib/python3.5/site-packages/pluggy/callers.py", line 203, in _multicall
INTERNALERROR>     gen.send(outcome)
INTERNALERROR>   File "/home/aurzenligl/projects/pytest-logger/pytest_logger/plugin.py", line 112, in pytest_runtest_makereport
INTERNALERROR>     destdir_relpath = os.path.relpath(self._logsdir.join(nodeid), failedlogsdir.join(nodepath))
INTERNALERROR>   File "/home/aurzenligl/projects/pytest-logger/.tox/py35-pytest37/lib/python3.5/posixpath.py", line 459, in relpath
INTERNALERROR>     genericpath._check_arg_types('relpath', path, start)
INTERNALERROR>   File "/home/aurzenligl/projects/pytest-logger/.tox/py35-pytest37/lib/python3.5/genericpath.py", line 143, in _check_arg_types
INTERNALERROR>     (funcname, s.__class__.__name__)) from None
INTERNALERROR> TypeError: relpath() argument must be str or bytes, not 'LocalPath'
```